ReactWindows/ReactNative.Net46/ReactPage.cs

```diff
@@ -111,17 +129,26 @@ public void OnSuspend()
         /// </param>
         public void OnResume(Action onBackPressed)
         {
-            _reactInstanceManager.OnResume(onBackPressed);
+            this.ReactInstanceManager.OnResume(onBackPressed);
         }
 
         /// <summary>
         /// Called before the application shuts down.
         /// </summary>
         public Task DisposeAsync()
         {
-            RootView.RemoveHandler(Keyboard.KeyDownEvent, (KeyEventHandler)OnAcceleratorKeyActivated);
+            RootView?.RemoveHandler(Keyboard.KeyDownEvent, (KeyEventHandler) OnAcceleratorKeyActivated);
+
+            //NOTE: Don't want to create the reactInstanceManager to just dispose of it... what should
+            //be returned here if IsValueCreated == false?  Should this method be tagged with async
+            //and the this.ReactInstanceManager.DisposeAsync() have an await?
 
-            return _reactInstanceManager.DisposeAsync();
+            //if (_reactInstanceManager.IsValueCreated)
+            //{
+            //    return this.ReactInstanceManager.DisposeAsync();
+            //}
+
+            return this.ReactInstanceManager.DisposeAsync();
```

Comment:
The goal here was to check if the reactInstanceManager has been created before trying to Dispose it... The return statement is not correct because it means that we may be creating the object only to dispose of it (clearly not what we want)... 

The commented out code is the solution to the above problem, but I'm not sure what the correct return statement is for the case when IsValueCreated is false. 

The method is not marked as "async" and we are not awaiting on the DisposeAsync.  Should we mark the dispose async and await on the DisposeAsync?